pkg/utils/parse_arf_result.go

```diff
@@ -453,10 +484,66 @@ func remediationsFromString(scheme *runtime.Scheme, name string, namespace strin
 	return rems
 }
 
+func getQuotedStrings(s string) []string {
+	matchStr := regexp.MustCompile(`".*?"`).FindAllString(s, -1)
+	matchList := make([]string, len(matchStr))
+	for i, m := range matchStr {
+		matchList[i] = m[1 : len(m)-1]
+	}
+	return matchList
+}
+
+func parseValues(remContent string, resultValues map[string]string) (string, []string, []string, error) {
+
+	t, err := template.New("").Option("missingkey=error").Parse(remContent)
+	var valuesUsedList []string
+	var valuesMissingList []string
+	if err != nil {
+		return remContent, valuesUsedList, valuesMissingList, err //Error creating template
+	}
+	buf := &bytes.Buffer{}
+
+	errMsg := t.Execute(buf, resultValues) //Will return error if a value is not found in XCCDF
+	fixedText := buf.String()
+	if errMsg == nil {
+		valuesUsedList = getParsedValueName(t)
+		return fixedText, valuesUsedList, valuesMissingList, errMsg
+	} else if strings.Contains(errMsg.Error(), "map has no entry") {
+		valuesMissingList = getQuotedStrings(errMsg.Error())[1:] // [1:] to get ride of first empty quote
+		return fixedText, valuesUsedList, valuesMissingList, errMsg
+	}
+	return fixedText, valuesUsedList, valuesMissingList, errMsg
+}
+
+func getParsedValueName(t *template.Template) []string {
+	return ListTemplFields(t)
+}
+
+func ListTemplFields(t *template.Template) []string {
```

Comment:
I guess this function doesn't have to be exported.

Reply:
I will fix that